src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -190,7 +280,8 @@ internal static class MethodGenerator
                         tokens.Add(SyntaxFactory.Token(SyntaxKind.SealedKeyword));
                     }
 
-                    if (method.IsStatic)
+                    var preferStaticFunction = workspace.Options.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction).Value;
```

Comment:
Likely need to add tests to the old extract method for this change where the `PreferStaticLocalFunction` is set to false